src/tooltip.js

```diff
@@ -39,17 +39,39 @@ nv.models.tooltip = function() {
     ;
 
     /*
+     * If a parent of the container has a translate transformation, fix the
+     * positioning. The fixed position of the tooltip will be based starting
+     * at this element.
+     */
+    var fixTranslation = function(pos) {
+        var obj = chartContainer;
+
+        while(obj && obj.style) {
+            if(getComputedStyle(obj).transform != 'none') {
+                var rect = obj.getBoundingClientRect();
+                pos.left -= rect.left;
+                pos.top -= rect.top;
+                break;
+            }
+
+            obj = obj.parentNode;
```

Comment:
A less invasive approach would be to put the nvd3 chart and the tooltip
inside a new div, and add a translate(0,0) to the new div.  This would
ensure that adding the additional translation wouldn't break the rest of
the code of the user that depends on fixed positioning.

What do you think?

Em Qui, 26 de mai de 2016 12:04 PM, Renato Utsch <renatoutsch@gmail.com>
escreveu:

> It is always going to walk the object tree, but it will stop in the case
> of a translation. The problem is that if there is a parent that has a css
> translation, the fixed positioning of children will be based on the
> position of this parent, not the screen. And as d3.event.client* is based
> on the position of the screen, it has to take into account the translation
> of this parent. The complexity of walking up the tree is not good, I agree.
>
> I had a good idea.  We can just look at the parent of the tooltip, and if
> it doesn't have a translation property, we add a translate(0,0). We would
> only have to take into account the immediate parent of the tooltip and it
> would always work.
>
> I'll do this later today, thanks for the good idea @liquidpele.
>
> Em Qui, 26 de mai de 2016 11:23 AM, liquidpele <notifications@github.com>
> escreveu:
>
>> In src/tooltip.js
>> <https://github.com/novus/nvd3/pull/1674#discussion_r64754094>:
>>
>> > +     * If a parent of the container has a translate transformation, fix the
>> > +     * positioning. The fixed position of the tooltip will be based starting
>> > +     * at this element.
>> > +     */
>> > +    var fixTranslation = function(pos) {
>> > +        var obj = chartContainer;
>> > +
>> > +        while(obj && obj.style) {
>> > +            if(getComputedStyle(obj).transform != 'none') {
>> > +                var rect = obj.getBoundingClientRect();
>> > +                pos.left -= rect.left;
>> > +                pos.top -= rect.top;
>> > +                break;
>> > +            }
>> > +
>> > +            obj = obj.parentNode;
>>
>> This is always going to walk up the whole object tree every time it
>> adjusts the tooltip position... which is a lot... What was
>> getBoundingClientRect().top and left doing that was wrong exactly? From
>> what I can tell it is supposed to take translate into account, there's even
>> a bugzilla bug about it that they fixed at one point.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/novus/nvd3/pull/1674/files/7aef28714acf8368a01264f36ea7f6d6db95c912#r64754094>
>>
>